fgcm/fgcmGray.py

```diff
@@ -211,10 +211,10 @@ def computeExpGrayForInitialSelection(self):
 
         np.add.at(expGrayForInitialSelection,
                   obsExpIndex[goodObs],
-                  EGray[goodObs])
+                  (EGray[goodObs]).astype(expGrayForInitialSelection.dtype))
         np.add.at(expGrayRMSForInitialSelection,
                   obsExpIndex[goodObs],
-                  EGray[goodObs]**2.)
+                  (EGray[goodObs]).astype(expGrayRMSForInitialSelection.dtype)**2.)
```

Comment:
Should the `astype` also wrap the `**2.` here?

Reply:
The opposite, I think.  It should be promoted before squaring.

Reply:
Ok...you may want to look for other cases (I think I saw others where the promotion was after squaring).